diff --git a/node_modules/@bazel/esbuild/esbuild.bzl b/node_modules/@bazel/esbuild/esbuild.bzl
index 6df768a..24d761b 100755
--- a/node_modules/@bazel/esbuild/esbuild.bzl
+++ b/node_modules/@bazel/esbuild/esbuild.bzl
@@ -44,12 +44,11 @@ def _esbuild_impl(ctx):
 
     args.add("--bundle", entry_point.path)
     args.add("--sourcemap")
-    args.add("--keep-names")
     args.add_joined(["--platform", ctx.attr.platform], join_with = "=")
     args.add_joined(["--target", ctx.attr.target], join_with = "=")
     args.add_joined(["--log-level", "info"], join_with = "=")
     args.add_joined(["--metafile", metafile.path], join_with = "=")
-    args.add_all(ctx.attr.define, format_each = "--define=%s")
+    args.add_all(ctx.attr.define, format_each = "--define:%s")
     args.add_all(ctx.attr.external, format_each = "--external=%s")
 
     # disable the error limit and show all errors
