diff --git a/node_modules/@commitlint/rules/lib/scope-enum.js b/node_modules/@commitlint/rules/lib/scope-enum.js
index 7d06716..9af0457 100644
--- a/node_modules/@commitlint/rules/lib/scope-enum.js
+++ b/node_modules/@commitlint/rules/lib/scope-enum.js
@@ -31,7 +31,7 @@ const scopeEnum = (parsed, when = 'always', value = []) => {
     }
     // Scopes may contain slash or comma delimiters to separate them and mark them as individual segments.
     // This means that each of these segments should be tested separately with `ensure`.
-    const delimiters = /\/|\\|,/g;
+    const delimiters = /\\|,/g;
     const scopeSegments = parsed.scope.split(delimiters);
     const negated = when === 'never';
     const result = value.length === 0 ||
