diff --git a/lib/scope-enum.js b/lib/scope-enum.js
index 35469611dc4f7b95f749e0fe6653879d2fd6b393..4e079c68deefc0c6b22b34fde4dd16043d1cc690 100644
--- a/lib/scope-enum.js
+++ b/lib/scope-enum.js
@@ -12,11 +12,11 @@ export const scopeEnum = ({ scope }, when = 'always', value = []) => {
     const isScopeInEnum = (scope) => ensure.enum(scope, value);
     let isValid;
     if (when === 'never') {
-        isValid = !messageScopes.some(isScopeInEnum);
+        isValid = !messageScopes.some(isScopeInEnum) && !isScopeInEnum(scope);
         errorMessage.splice(1, 0, 'not');
     }
     else {
-        isValid = messageScopes.every(isScopeInEnum);
+        isValid = messageScopes.every(isScopeInEnum) || isScopeInEnum(scope);
     }
     return [isValid, message(errorMessage)];
 };
