diff --git a/dist/conventional-commits/get-changelog-config.js b/dist/conventional-commits/get-changelog-config.js
index bb2dce0e7fab0faa5d64f250a247108a89a41a3b..d2ecc855efdb42c386580e334bf25616074fddd2 100644
--- a/dist/conventional-commits/get-changelog-config.js
+++ b/dist/conventional-commits/get-changelog-config.js
@@ -4,8 +4,9 @@ import npa from 'npm-package-arg';
 import pify from 'pify';
 function flattenConfigResult(config) {
     const flatConfig = config?.conventionalChangelog || config;
-    flatConfig.parser = flatConfig.parserOpts || flatConfig.parser;
+    flatConfig.parser = config.recommendedBumpOpts?.parserOpts || flatConfig.parserOpts || flatConfig.parser;
     flatConfig.writer = flatConfig.writerOpts || flatConfig.writer;
+    flatConfig.whatBump = config.recommendedBumpOpts?.whatBump || flatConfig.whatBump;
     return flatConfig;
 }
 export class GetChangelogConfig {
