diff --git a/lib/index.d.ts b/lib/index.d.ts
index 5ec0eb759afa8af975a26cdbd4526871e2aba38b..edda5a8e3264ac8c7fe023e3a83bac1d1339311b 100644
--- a/lib/index.d.ts
+++ b/lib/index.d.ts
@@ -39,5 +39,5 @@ export type Props = {
     disableParentContext?: boolean | null | undefined;
 };
 import type { MDXComponents } from 'mdx/types.js';
-import React from 'react';
+import * as React from 'react';
 //# sourceMappingURL=index.d.ts.map
