diff --git a/lib/compat.cjs b/lib/compat.cjs
new file mode 100644
index 0000000000000000000000000000000000000000..b32f62fad3890b80b5d68291dda88d7799f97dd3
--- /dev/null
+++ b/lib/compat.cjs
@@ -0,0 +1 @@
+module.exports = require("@typescript/typescript6");
diff --git a/lib/compat.d.cts b/lib/compat.d.cts
new file mode 100644
index 0000000000000000000000000000000000000000..5e978dee1da2b60eabca0520ae4aac2c28829dbf
--- /dev/null
+++ b/lib/compat.d.cts
@@ -0,0 +1,2 @@
+import ts = require("@typescript/typescript6");
+export = ts;
diff --git a/package.json b/package.json
index f4242a69df961fcaf5210ac08ace176ced18c6a4..5302e15bc1ac890e04d4e2420bd462a86ae7cf54 100644
--- a/package.json
+++ b/package.json
@@ -36,7 +36,7 @@
     },
     "exports": {
         "./package.json": "./package.json",
-        ".": "./lib/version.cjs",
+        ".": "./lib/compat.cjs",
         "./unstable/sync": "./dist/api/sync/api.js",
         "./unstable/async": "./dist/api/async/api.js",
         "./unstable/fs": "./dist/api/fs.js",
