diff --git a/MODULE.bazel b/MODULE.bazel
index ef4c5d2..6a12951 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -5,6 +5,7 @@ module(
 )
 
 bazel_dep(name = "rules_rust", version = "0.69.0")
+bazel_dep(name = "rules_rs", version = "0.0.58")
 bazel_dep(name = "bazel_features", version = "1.32.0")
 bazel_dep(name = "bazel_skylib", version = "1.8.2")
 bazel_dep(name = "platforms", version = "1.0.0")
diff --git a/private/transitions.bzl b/private/transitions.bzl
index 7917c54..8e40651 100644
--- a/private/transitions.bzl
+++ b/private/transitions.bzl
@@ -12,7 +12,12 @@ def _wasm_bindgen_transition(_settings, attr):
     Returns:
         dict: A dict of new build settings values to apply
     """
-    return {"//command_line_option:platforms": str(Label("@rules_rust//rust/platform:{}".format(attr.target_arch)))}
+    if attr.target_arch == "wasm32":
+        platform = "@rules_rs//rs/experimental/platforms:wasm32-unknown-unknown"
+    else:
+        platform = "@rules_rust//rust/platform:{}".format(attr.target_arch)
+
+    return {"//command_line_option:platforms": str(Label(platform))}
 
 wasm_bindgen_transition = transition(
     implementation = _wasm_bindgen_transition,
diff --git a/private/wasm_bindgen.bzl b/private/wasm_bindgen.bzl
index 1876dd9..1363355 100644
--- a/private/wasm_bindgen.bzl
+++ b/private/wasm_bindgen.bzl
@@ -161,10 +161,6 @@ WASM_BINDGEN_ATTR = {
     "wasm_file": attr.label(
         doc = "The `.wasm` crate to generate bindings for.",
         allow_single_file = True,
-        providers = [
-            [rust_common.test_crate_info],
-            [rust_common.crate_info],
-        ],
         aspects = [
             rust_analyzer_aspect,
             rustfmt_aspect,
