diff --git a/node_modules/make-plural-compiler/lib/parser.js b/node_modules/make-plural-compiler/lib/parser.js
index 63be531..b4d4bf6 100644
--- a/lib/parser.js
+++ b/lib/parser.js
@@ -6,6 +6,10 @@ Object.defineProperty(exports, "__esModule", {
 exports.default = void 0;
 
 class Parser {
+  /**
+   * plural condition
+   * @param {String} cond
+   */
   parse(cond) {
     if (cond === 'i = 0 or n = 1') return 'n >= 0 && n <= 1';
     if (cond === 'i = 0,1') return 'n >= 0 && n < 2';
@@ -15,7 +19,10 @@ class Parser {
       return 'n == 1 && v0';
     }
 
-    return cond.replace(/([^=\s])([!=%]+)([^=\s])/g, '$1 $2 $3').replace(/([tv]) (!?)= 0/g, (m, sym, noteq) => {
+    return cond.replace(/([^=\s])([!=%]+)([^=\s])/g, '$1 $2 $3').replace(/[ce]\s?=\s?(\d+)/g, (m, c) => {
+      this.c = c;
+      return m;
+    }).replace(/([tv]) (!?)= 0/g, (m, sym, noteq) => {
       const sn = sym + '0';
       this[sn] = 1;
       return noteq ? '!' + sn : sn;
@@ -67,7 +74,14 @@ class Parser {
     }
 
     if (!vars.length) return '';
-    return 'var ' + ["s = String(n).split('.')", ...vars].join(', ');
+
+    if (this.c) {
+      vars.unshift('_n = String(n)', 'se = _n.split(/[ce]/)', 'e = se[1] || 0', 'c = e', 's = String(e ? Number(se[0]) * Math.pow(10, e) : _n).split(".")');
+    } else {
+      vars.unshift("s = String(n).split('.')");
+    }
+
+    return 'var ' + vars.join(', ');
   }
 
 }
diff --git a/node_modules/make-plural-compiler/lib/tests.js b/node_modules/make-plural-compiler/lib/tests.js
index 967ffb2..768636c 100644
--- a/lib/tests.js
+++ b/lib/tests.js
@@ -19,10 +19,11 @@ class Tests {
     };
   }
 
-  error(n, type, msg) {
+  error(n, type, msg, fn) {
     const lc = JSON.stringify(this.lc);
     const val = JSON.stringify(n);
-    return new Error(`Locale ${lc} ${type} rule self-test failed for ${val} (${msg})`);
+    return new Error(`Locale ${lc} ${type} rule self-test failed for ${val} (${msg}). Function:
+${fn}`);
   }
 
   testCond(n, type, expResult, fn) {
@@ -30,13 +31,13 @@ class Tests {
       var r = fn(n, type === 'ordinal');
     } catch (error) {
       /* istanbul ignore next: should not happen unless CLDR data is broken */
-      throw this.error(n, type, error);
+      throw this.error(n, type, error, fn.toString());
     }
 
     if (r !== expResult) {
       const res = JSON.stringify(r);
       const exp = JSON.stringify(expResult);
-      throw this.error(n, type, `was ${res}, expected ${exp}`);
+      throw this.error(n, type, `was ${res}, expected ${exp}`, fn.toString());
     }
 
     return true;
@@ -51,7 +52,7 @@ class Tests {
 
     data.values.forEach(n => {
       this.testCond(n, type, cat, fn);
-      if (!/\.0+$/.test(n)) this.testCond(Number(n), type, cat, fn);
+      if (!/\.0+$/.test(n)) this.testCond(n, type, cat, fn);
     });
     return true;
   }
